pkg/loki/modules.go

```diff
@@ -178,6 +178,52 @@ func (t *Loki) stopStore() error {
 	return nil
 }
 
+// listDeps recursively gets a list of dependencies for a passed moduleName
+func listDeps(m moduleName) []moduleName {
+	deps := modules[m].deps
+	for _, d := range modules[m].deps {
+		deps = append(deps, listDeps(d)...)
+	}
+	return deps
+}
+
+// orderedDeps gets a list of all dependencies ordered so that items are always after any of their dependencies.
+func orderedDeps(m moduleName) []moduleName {
+	deps := listDeps(m)
+
+	// get a unique list of moduleNames, with a flag for whether they have been added to our result
+	uniq := map[moduleName]bool{}
+	for _, dep := range deps {
+		uniq[dep] = false
+	}
+
+	result := make([]moduleName, 0)
```

Comment:
Lets do `[]moduleName{}` or `make([]moduleName, 0, len(uniq)`.  Anything but this!